include/opt-sched/Scheduler/lnkd_lst.h

```diff
@@ -818,7 +819,12 @@ inline T *PriorityList<T, K>::GetNxtPriorityElmnt() {
 
 template <class T, class K>
 inline T *PriorityList<T, K>::GetNxtPriorityElmnt(K &key) {
+  if (LinkedList<T>::wasTopRmvd_) {
```

Comment:
Should fall into existing handling

`  bool rtrvInvalid = LinkedList<T>::wasTopRmvd_ || LinkedList<T>::itrtrReset_;
  assert(rtrvInvalid || LinkedList<T>::rtrvEntry_ != NULL);

  if (rtrvInvalid) {
    LinkedList<T>::rtrvEntry_ = LinkedList<T>::topEntry_;
    LinkedList<T>::itrtrReset_ = false;
    LinkedList<T>::wasTopRmvd_  = false;
  } else {
    LinkedList<T>::rtrvEntry_ = LinkedList<T>::rtrvEntry_->GetNext();
  }
`